sdk/metric/internal/aggregate/sum_test.go

```diff
@@ -167,118 +168,92 @@ func TestDeltaSumReset(t *testing.T) {
 func TestPreComputedDeltaSum(t *testing.T) {
 	var mono bool
 	agg := NewPrecomputedDeltaSum[int64](mono)
-	require.Implements(t, (*precomputeAggregator[int64])(nil), agg)
+	require.Implements(t, (*Aggregator[int64])(nil), agg)
 
 	attrs := attribute.NewSet(attribute.String("key", "val"))
 	agg.Aggregate(1, attrs)
-	got := agg.Aggregation()
 	want := metricdata.Sum[int64]{
 		IsMonotonic: mono,
 		Temporality: metricdata.DeltaTemporality,
 		DataPoints:  []metricdata.DataPoint[int64]{point[int64](attrs, 1)},
 	}
 	opt := metricdatatest.IgnoreTimestamp()
-	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+	metricdatatest.AssertAggregationsEqual(t, want, agg.Aggregation(), opt)
 
-	// No observation means no metric data
-	got = agg.Aggregation()
-	metricdatatest.AssertAggregationsEqual(t, nil, got, opt)
+	// No observation results in an empty aggregation, and causes previous
+	// observations to be forgotten.
+	metricdatatest.AssertAggregationsEqual(t, nil, agg.Aggregation(), opt)
 
-	agg.(precomputeAggregator[int64]).aggregateFiltered(1, attrs)
-	got = agg.Aggregation()
-	// measured(+): 1, previous(-): 1, filtered(+): 1
+	agg.Aggregate(1, attrs)
+	// measured(+): 1, previous(-): 0
 	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 1)}
-	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
-
-	// Filtered values should not persist.
-	got = agg.Aggregation()
-	// No observation means no metric data
-	metricdatatest.AssertAggregationsEqual(t, nil, got, opt)
+	metricdatatest.AssertAggregationsEqual(t, want, agg.Aggregation(), opt)
 
-	// Override set value.
+	// Duplicate observations add
 	agg.Aggregate(2, attrs)
 	agg.Aggregate(5, attrs)
-	// Filtered should add.
-	agg.(precomputeAggregator[int64]).aggregateFiltered(3, attrs)
-	agg.(precomputeAggregator[int64]).aggregateFiltered(10, attrs)
-	got = agg.Aggregation()
-	// measured(+): 5, previous(-): 0, filtered(+): 13
-	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 18)}
-	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+	agg.Aggregate(3, attrs)
+	agg.Aggregate(10, attrs)
+	// measured(+): 20, previous(-): 1
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, 19)}
+	metricdatatest.AssertAggregationsEqual(t, want, agg.Aggregation(), opt)
 
 	// Filtered values should not persist.
 	agg.Aggregate(5, attrs)
-	got = agg.Aggregation()
-	// measured(+): 5, previous(-): 18, filtered(+): 0
-	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, -13)}
-	metricdatatest.AssertAggregationsEqual(t, want, got, opt)
+	// measured(+): 5, previous(-): 20
+	want.DataPoints = []metricdata.DataPoint[int64]{point[int64](attrs, -15)}
+	metricdatatest.AssertAggregationsEqual(t, want, agg.Aggregation(), opt)
```

Comment:
There is no distinction between filtered and non-filtered at this point. It seems like this section could just be removed as it will likely cause future confusion.

Reply:
removed